### PR TITLE
Feature/sckan-279 - Admin interface changes for region + layer

### PR DESCRIPTION
**Pull request by D-GopalKrishna**
Jira ticket - https://metacell.atlassian.net/browse/SCKAN-279

AE -> AnatomicalEntity

Task description:

- ~Allow to add Synonyms from AnatomicalEntityMeta and AnatomicalEntityIntersections.~ 
- ~Added a custom form to this. And allow a charfield with comma separated values as synonyms.~ 
- ~This then triggers a post_save - django signal that saves Synonyms if it is passed along while saving.~

<br>

- Add Synonyms to AnatomicalEntity in another tab. In AE - we can also create new simple entities (AEMeta) or RegionLayer Combinations, and link synonyms to them to create a AE.
- Can create a region layer combination (but they are not AE)
- Can create a region or a layer (But they are not AE - they are AEMeta)

<br>
<br>

Acceptance criteria (from the Jira Ticket):
- allow the admins to add a region through the admin interface - ✅ - already available.
- allow the admins to add a layer through the admin interface - ✅ - already available.
- allow the admins to add/edit a synonym for an anatomical entity (region, layer) - 🟠 - Done in this PR. Changes being - since Synonyms are connected to AnatomicalEntity - We cannot directly have synonyms for Simple AE/Region/Layer. Rather we have  a custom form to get synonyms for AEIntersections and Simple Entities (AEMeta) and save them in AnatomicalEntity. 

- allow the admins to add an intersection region+layer through the admin interface - ✅ - already available.
- allow the admins to pick for origins, vias, destinations either a simple anatomical entity or an intersection region+layer. - ✅ - already available - Users can pick the Anantomical Entities in origins, vias, destinations.

### CODE REVIEW

**Comment by D-GopalKrishna** [comment]
@zsinnema we do need this -> 

```
def get_model_perms(self, request):
        return {}
```

to register the AnatomicalEntityMetaAdmin to be referenced by AnatomicalEntityAdmin.autocomplete_fields.

File: backend/composer/admin.py [hunk 1]
```diff
@@ -89,26 +97,32 @@ class SentenceAdmin(
...
+        return ', '.join([synonym.name for synonym in synonyms])
 
 
 class AnatomicalEntityMetaAdmin(admin.ModelAdmin):
     list_display = ("name", "ontology_uri")
     list_display_links = ("name", "ontology_uri")
-    search_fields = ("name",)
+    search_fields = ("name", "ontology_uri")
+
+    def get_model_perms(self, request):
```
Comment (zsinnema) [inline backend/composer/admin.py:124]: question: @D-GopalKrishna why is this needed? can't we just remove the register (at the bottom)?

Comment (D-GopalKrishna) [reply backend/composer/admin.py:124]: No, django admin requires it if we need to use - `autocomplete_fields`

image
